zulip/zulip/__init__.py

```diff
@@ -226,12 +226,18 @@ def init_from_options(options: Any, client: Optional[str] = None) -> 'Client':
         client = options.zulip_client
     elif client is None:
         client = _default_client()
-    return Client(email=options.zulip_email, api_key=options.zulip_api_key,
-                  config_file=options.zulip_config_file, verbose=options.verbose,
-                  site=options.zulip_site, client=client,
-                  cert_bundle=options.cert_bundle, insecure=options.insecure,
-                  client_cert=options.client_cert,
-                  client_cert_key=options.client_cert_key)
+
+    new_client = Client(email=options.zulip_email, api_key=options.zulip_api_key,
+                        config_file=options.zulip_config_file, verbose=options.verbose,
+                        site=options.zulip_site, client=client,
+                        cert_bundle=options.cert_bundle, insecure=options.insecure,
+                        client_cert=options.client_cert,
+                        client_cert_key=options.client_cert_key)
+
+    if new_client.get_profile()["result"] == "error":
```

Comment:
Awesome, this sounds like the code we want. Just one thing I'd change here: we have some more places that also call `Client()` to create a new client. (`git grep 'Client(' | wc -l` shows 31 cases, most of which are calls to `zulip.Client()`. We also want to throw this error in thjose cases, thus, we should have this block of code in the Client class, just before it returns the client object.

Reply:
I thought 0 meant an error. does -1 work in this case?